src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorTagHelperBinderPhase.cs

```diff
@@ -397,28 +444,42 @@ internal static bool IsTypeInScope(StringSegment typeName, string currentNamespa
             // open file in the editor. We mangle the class name for its generated code, so using that here to filter these out.
             internal static bool IsTagHelperFromMangledClass(TagHelperDescriptor tagHelper)
             {
-                StringSegment typeName = tagHelper.GetTypeName();
+                StringSegment className;
                 if (tagHelper.IsChildContentTagHelper())
                 {
                     // If this is a child content tag helper, we want to look at it's original type.
                     // E.g, if the type name is `Test.__generated__MyComponent.ChildContent`, we want to look at `Test.__generated__MyComponent`.
-                    TrySplitNamespaceAndType(typeName, out var namespaceTextSpan, out var _);
-                    typeName = GetTextSpanContent(namespaceTextSpan, typeName);
+                    TrySplitNamespaceAndType(tagHelper, out var typeNamespace);
+                    return TrySplitNamespaceAndType(typeNamespace, out var _, out className)
+                        && ComponentMetadata.IsMangledClass(className);
                 }
-                if (!TrySplitNamespaceAndType(typeName, out var _, out var classNameTextSpan))
+
+                return TrySplitNamespaceAndType(tagHelper, out var _, out className) &&
+                    ComponentMetadata.IsMangledClass(className);
+            }
+
+            internal static bool TrySplitNamespaceAndType(TagHelperDescriptor tagHelperDescriptor, out StringSegment @namespace)
+                => TrySplitNamespaceAndType(tagHelperDescriptor, out @namespace, out _);
+
+            internal static bool TrySplitNamespaceAndType(TagHelperDescriptor tagHelperDescriptor, out StringSegment @namespace, out StringSegment typeName)
+            {
+                if (tagHelperDescriptor.ParsedTypeInfo is { } value)
```

Comment:
woa what's the `{ }` syntax. Anonymous type?

Reply:
It's pattern matching that says I match any non-null value. It's kinda a stop-in for not being able to say 

```
if (blah is not null value)
```

Reply:
Deffffinitely going to have to get used to that one. Not sure how I feel about it 😄

Reply:
Can't you also do `if (tagHelperDescriptor.ParsedTypeInfo is ParsedTypeInformation value)` here as well or is that asserting something different?

Reply:
You can. But then I have to type stuff and lose out on :100: points.